next/components/molecules/SearchPageNew/GeneralSearchResults.tsx

```diff
@@ -0,0 +1,97 @@
+import { Typography } from '@bratislava/component-library'
+import Pagination from '@bratislava/ui-bratislava/Pagination/Pagination'
+import SearchCardComposed from '@components/molecules/SearchPageNew/SearchCardComposed'
+import {
+  getDataBySearchOptionKey,
+  SearchFilters,
+} from '@components/molecules/SearchPageNew/searchDataFetchers'
+import { SearchResultsHeader } from '@components/molecules/SearchPageNew/SearchResultsHeader'
+import { SearchOption } from '@components/pages/searchPageContentNew'
+import { isDefined } from '@utils/isDefined'
+import { useTranslations } from 'next-intl'
+
+type GeneralSearchResultsProps = {
+  filters: SearchFilters
+  variant: 'allResults' | 'specificResults'
+  searchOption: SearchOption
+  handleShowMore?: React.Dispatch<React.SetStateAction<Set<SearchOption['id']>>>
+  handlePageChange?: React.Dispatch<React.SetStateAction<number>>
```

Comment:
In props, we use `on...` instead of `handle...` for function names.
`handleShowMore` --> `onShowMore`
`handlePageChange` --> `onPageChange`.

Please import `Dispatch` and `SetStateAction`.
`React.Dispatch<React.SetStateAction<number>>` --> `Dispatch<SetStateAction<number>>`

You can also use the same type for `onPageChange` as it is used in`<Pagination>`